<launch>

    <arg name="global_laser_cloud_topic"    default="/global_laser_cloud_map"/>
    <arg name="global_map_load_path_"       default="$(find grid_cost_map)/map/carla_map.png"/> <!-- 如果该路径不为空，则尝试从该路径下预加载png的代价地图 -->
    <arg name="global_map_res_"             default="0.4"/>
    <arg name="global_map_save_flag_"       default="false"/>
    <arg name="global_map_save_path_"       default="$(find grid_cost_map)/map/carla_map.png"/>

    <arg name="local_laser_cloud_topic"     default="/all_lidars_preprocessed_ros"/>
    <arg name="local_map_res_"              default="0.3"/>
    <arg name="local_map_length_"           default="30"/>


    <node pkg="grid_cost_map" type="build_cost_map" name="grid_cost_map" output="screen">
        <param name="global_laser_cloud_topic"  type="str"      value="$(arg global_laser_cloud_topic)"/>
        <param name="global_map_load_path_"     type="str"      value="$(arg global_map_load_path_)"/>
        <param name="global_map_res_"           type="double"   value="$(arg global_map_res_)"/>
        <param name="global_map_save_flag_"     type="bool"     value="$(arg global_map_save_flag_)"/>
        <param name="global_map_save_path_"     type="str"      value="$(arg global_map_save_path_)"/>
        <rosparam command="load" file="$(find grid_cost_map)/config/global_map_filter_chain.yaml" />
    
        <param name="local_laser_cloud_topic"   type="str"      value="$(arg local_laser_cloud_topic)"/>
        <param name="local_map_res_"            type="double"   value="$(arg local_map_res_)"/>
        <param name="local_map_length_"         type="double"   value="$(arg local_map_length_)"/>
        <rosparam command="load" file="$(find grid_cost_map)/config/local_map_filter_chain.yaml" />
    </node>

</launch>
